.vscode/settings.json

```diff
@@ -0,0 +1,13 @@
+{
+    // NOTE (aseipp): don't autoformat {markdown, toml, et cetera} files, since
+    // there's no real standardized format we've ever tried to agree on; until
+    // then, such a feature tends to make diffs much worse and various people's
+    // editors config may have some "helpful" plugins that autoformat these
+    // filetypes, so this disables a footgun for everyone.
+    "[markdown]": {
+        "editor.formatOnSave": false
+    },
+    "[toml]": {
+        "editor.formatOnSave": false
+    }
+}
```

Comment:
My preferred (but imperfect) solution to such a usecase would be to have you put .vscode into a global gitignore or .git/exclude and have your own vscode config.

On problem I see with this is that finding the per-repo `.git/exclude` with jj is tricky with non-colocated repos (I'm not even sure it's supported).